<?xml version="1.0"?>
<launch>
  <arg name="robot_name" default="default_robot" />
  <arg name="mode" default="none" />
  <arg name="target" default="localhost" />

<!--
  <arg  name="rvizconfig" default="$(find s4_launch)/rviz/default.rviz" />
-->
  <arg name="rvizconfig" default="$(find s4_description)/rviz/robot_test.rviz" />

  <group ns="$(arg robot_name)/operation">
    <node pkg="joy" type="joy_node" name="joy_node" >
    </node>  

    <node pkg="s4_operation" type="s4_operation_joy_converter" name="s4_operation_joy_converter">
      <param name="joy_type" value="ps3" />
    </node>

    <node pkg="s4_operation" type="s4_operation_test_app" name="s4_operation_test_app" >
      <remap from="move_action" to="/$(arg robot_name)/navigation/move_action" />    
      <remap from="turret_action" to="/$(arg robot_name)/firecontrol/turret_action" />
      <remap from="select_action" to="/$(arg robot_name)/selection/select_action" />

      <remap from="objects" to="/$(arg robot_name)/detection/objects" />
    </node>

    <node pkg="s4_operation" type="image_overlay.py" name="image_overlay" output="screen">
      <param name="tf_prefix" value="$(arg robot_name)" />
      <remap from="input_image" to="/$(arg robot_name)/detection/image_rect_color" />            
      <remap from="input_objects" to="/$(arg robot_name)/detection/objects" />            
      <remap from="focus" to="/$(arg robot_name)/selection/focus" />
      <remap from="scan" to="/$(arg robot_name)/device/sensor0/front_laser/scan_filtered" />
      <remap from="voltage" to="/$(arg robot_name)/device/body/battery/s4_body_battery/voltage" />
      <remap from="point" to="/$(arg robot_name)/device/gun0/command" />
      <remap from="camera_info" to="/$(arg robot_name)/device/sensor0/head_camera/camera_info" />
      <param name="rpy_offset" value="/$(arg robot_name)/device/sensor0/head_camera/s4_hardware_call_camera_info/rpy_offset"/>
    </node>
    <node name="image_view" pkg="image_view" type="image_view" >
      <remap from="image" to="view_image"/>
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

    <!-- web -->
    <include file="$(find roswww)/launch/roswww.launch" />
    <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" />
    <node pkg="web_video_server" type="web_video_server" name="web_video_server">
      <param name="quality" value="10" />
    </node>
    <node name="image_republish" pkg="image_transport" type="republish" args="raw compressed">
      <remap from="in" to="/s4n1/operation/image_view/output" />
      <remap from="out" to="/web/image_raw" />
    </node>
    <node pkg="s4_operation" type="s4_operation_joy_converter" name="s4_operation_joy_converter_web">
      <remap from="joy" to="/web/joy" />
      <param name="joy_type" value="iphone" />
      <param name="restamp" value="true" />
    </node>

  </group>


</launch>
